### DISCUSSION TITLE
Find numerical differences between numerical values

### DISCUSSION THREAD

**Discussion by phourglass** [WinMerge/winmerge#1851] [Q&A]
Hello,

I use winmerge to compare relatively large text files which include a lot of numerical values (models).
The format is quite complex but works well for line by line comparison as it has keywords in it besides numerical values.

The files sometimes change due to the software writing them and:
* rounding errors: 0.12345678 becomes 0.1234568
* use of different formats: 0.1 becomes 0.100000 or 1E-1

The files are the same or similar from a numerical standpoint but it is tricky to find out when 1E6 lines differ due to rounding.

There are command lines tools that can partially do that but have limitations are are not as nice as winmerge:
https://github.com/quinoacomputing/ndiff
https://www.math.utah.edu/~beebe/software/ndiff/

Is there a way to do that in winmerge?
I saw there are plugins (I am ignorant about it).
Would there be a way to interface such tools as plugins?

Thanks
Philippe

**Comment by sdottaka** [comment]
Unfortunately, WinMerge doesn't have ndiff like functionality.

I don't know if it will work well with the files you are comparing, but I created and attached a plug-in that converts strings that can be regarded as numbers in text files to single-precision real numbers and compares them. 

You can copy the IgnoreNumberFormat.sct file in the attached IgnoreNumberFormat.zip to `C:\Program Files\WinMerge\MergePlugins` folder, start WinMerge, and apply it as follows.

![image](https://github.com/WinMerge/winmerge/assets/98126/b2f56768-36be-4a4c-9a11-00ee047bf94c)

IgnoreNumberFormat.sct is written in VBScript, so please correct it if it doesn't work.

[IgnoreNumberFormat.zip](https://github.com/WinMerge/winmerge/files/11478444/IgnoreNumberFormat.zip)
